tools/google-cloud-support-slackbot/google_cloud_support_slackbot.py

```diff
@@ -0,0 +1,811 @@
+        client.chat_postEphemeral(channel=channel_id, user=user_id, text=f"Case {case} could not be found in your org. If this case was recently created, please give the system 60 seconds to fetch it. Otherwise, double check your case number or confirm the org being tracked with your Slack admin.")
+    else:
+        tracker = {
+            "channel_id": channel_id, 
+            "case": case,
+            "channel_name": channel_name
+        }
+
+        exists = False
+    
+        for tc in tracked_cases:
+            if tc['channel_id'] == channel_id and tc['case'] == case:
+                exists = True
+                break
+
+        if exists == False:
+            tracked_cases.append(tracker)
+            
+            try:
+                file_overwrite(tracked_cases_file, tracked_cases)
+            except Exception as e:
+                logging.error(e, ' : {}'.format(datetime.now()))
+                time.sleep(0.1)
+                file_overwrite(tracked_cases_file, tracked_cases)
+
+            client.chat_postMessage(channel=channel_id, text=f"{channel_name} is now tracking case {case}")
+        else:
+            client.chat_postEphemeral(channel=channel_id, user=user_id, text=f"Case {case} is already being tracked in {channel_name}")
+
+
+
+def add_comment(channel_id, case, comment, user_id, user_name):
+    """
+    Add a comment to a Google Cloud support case.
+    
+    Parameters
+    ----------
+    channel_id : str
+        unique string used to idenify a Slack channel. Used to send messages to the channel
+    case : str
+        unique id of the case
+    comment : str
+        comment to be added to the case
+    user_id : str
+        the Slack user_id of the user who submitted the request. Used to send ephemeral
+        messages to the user
+    user_name : str
+        Slack user_name of the user that ran the command. Appended to the end of the
+        comment to identify who submitted submitted it, otherwise all comments will
+        show as coming from the case creator
+    """
+
+    client.chat_postEphemeral(channel=channel_id, user=user_id, text="Your request is processing ...")
+    parent = get_parent(case)
+    
+    if parent == 'Case not found':
+        client.chat_postEphemeral(channel=channel_id, user=user_id, text=f"Case {case} could not be found in your org. If this case was recently created, please give the system 60 seconds to fetch it. Otherwise, double check your case number or confirm the org being tracked with your Slack admin.")
+    else:
+        req_body = {
+            "body" : comment + '\n*Comment submitted by {} via Google Cloud Support Slack bot*'.format(user_name)
+        }
+        try:
+            req = support_service.cases().comments().create(parent=parent, body=req_body)
+            req.execute()
+            client.chat_postEphemeral(channel=channel_id, user=user_id, text=f"You added a new comment on case {case}: {comment}")
+        except Exception as e:
```

Comment:
What specific exception are you capturing here? Please narrow it down to the specific situations that you are trying to handle, this is too broad.

Reply:
This is an error I'm still trying pin down. The impact in this part of the code isn't as big, but in the case_updates() loop, it breaks the loop which stops updates from being fed. I've updated so it only does the try on the execute step where the failure occurs, but I still don't have the specifics on the error that is being raised